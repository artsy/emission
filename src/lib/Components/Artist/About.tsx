import React from "react"
import { createFragmentContainer, graphql } from "react-relay"

import { Dimensions, StyleSheet, View } from "react-native"

import NavButton from "../Buttons/NavigationButton"
import RelatedArtists from "../RelatedArtists"
import Separator from "../Separator"
import Articles from "./Articles"
import Biography from "./Biography"

import { About_artist } from "__generated__/About_artist.graphql"

interface Props {
  artist: About_artist
}

class About extends React.Component<Props> {
  render() {
    return (
      <View>
        {this.biography()}
        {this.articles()}
        {this.relatedArtists()}
      </View>
    )
  }

  biography() {
    if (this.props.artist.has_metadata) {
      return (
        <View>
          <Biography artist={this.props.artist as any} />
          {this.auctionResults()}
          <Separator style={styles.sectionSeparator} />
        </View>
      )
    }
  }

  auctionResults() {
    if (this.props.artist.is_display_auction_link) {
      // Keeps the same margins as the bio text
      const sideMargin = Dimensions.get("window").width > 700 ? 50 : 20
      const url = `/artist/${this.props.artist.gravityID}/auction-results`
      return (
        <NavButton title="Auction Results" href={url} style={{ marginLeft: sideMargin, marginRight: sideMargin }} />
      )
    }
  }

  articles() {
    if (this.props.artist.articles.length) {
      return (
        <View>
          <Articles articles={this.props.artist.articles as any} />
          <Separator style={styles.sectionSeparator} />
        </View>
      )
    }
  }

  relatedArtists() {
    return this.props.artist.related_artists.length ? (
      <RelatedArtists artists={this.props.artist.related_artists as any} />
    ) : null
  }
}

const styles = StyleSheet.create({
  sectionSeparator: {
    marginBottom: 20,
  },
})

export default createFragmentContainer(About, {
  artist: graphql`
    fragment About_artist on Artist {
      has_metadata
      is_display_auction_link
      slug
      ...Biography_artist
      related_artists: artists(size: 16) {
        ...RelatedArtists_artists
      }
      articles {
        ...Articles_articles
      }
    }
  `,
})
