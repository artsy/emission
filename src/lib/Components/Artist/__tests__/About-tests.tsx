import "react-native"

import * as React from "react"
import * as renderer from "react-test-renderer"

import About from "../about"

it("renders properly", () => {
  const artist = {
    has_metadata: true,
    is_display_auction_link: true,
    articles: [],
    related_artists: [],
  }
  const about = renderer.create(<About artist={artist} />).toJSON()
  expect(about).toMatchSnapshot()
})
