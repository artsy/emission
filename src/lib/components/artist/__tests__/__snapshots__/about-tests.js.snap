exports[`test renders properly 1`] = `
<View>
  <View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 20,
            "marginLeft": 20,
            "marginRight": 20,
          },
          Object {
            "marginLeft": 20,
            "marginRight": 20,
          },
        ]
      }>
      <TouchableWithoutFeedback
        onPress={[Function]}>
        <View
          style={Object {}}>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#e5e5e5",
                  "height": 1,
                  "marginLeft": -20,
                  "marginRight": -20,
                },
                Object {
                  "marginLeft": 0,
                  "marginRight": 0,
                },
              ]
            } />
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontFamily": "Avant Garde Gothic ITCW01Dm",
                  "marginBottom": 14,
                  "marginTop": 14,
                }
              }>
              AUCTION RESULTS
            </Text>
            <Image
              source={1}
              style={
                Object {
                  "alignSelf": "center",
                }
              } />
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#e5e5e5",
                  "height": 1,
                  "marginLeft": -20,
                  "marginRight": -20,
                },
                Object {
                  "marginLeft": 0,
                  "marginRight": 0,
                },
              ]
            } />
        </View>
      </TouchableWithoutFeedback>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#e5e5e5",
            "height": 1,
            "marginLeft": -20,
            "marginRight": -20,
          },
          Object {
            "marginBottom": 20,
          },
        ]
      } />
  </View>
</View>
`;
